internal/transformer/classes/generators/dockerfilegenerators/PHPdockerfilegenerator.go

```diff
@@ -0,0 +1,207 @@
+			return port, confFileName, err
+		}
+		tokens := strings.Split(match.Content, ":")
+		if len(tokens) > 1 {
+			port, err := strconv.ParseInt(tokens[1], 10, 32)
+			if err != nil {
+				logrus.Errorf("Error while converting the entered port from string to int : %s", err)
+				return int32(port), confFileName, err
+			}
+			return int32(port), confFileName, nil
+		}
+		defer confFile.Close()
+	}
+	return port, confFileName, err
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *PHPDockerfileGenerator) DirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	dirEntries, err := os.ReadDir(dir)
+	if err != nil {
+		logrus.Errorf("Error while trying to read directory : %s", err)
+		return nil, nil, err
+	}
+	for _, de := range dirEntries {
+		ext := filepath.Ext(de.Name())
+		if ext == phpExt {
+			unnamedServices = []transformertypes.TransformerPlan{{
+				Mode:              t.Config.Spec.Mode,
+				ArtifactTypes:     []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+				BaseArtifactTypes: []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+				Paths: map[string][]string{
+					artifacts.ProjectPathPathType: {dir},
+				},
+			}}
+			return nil, unnamedServices, nil
+		}
+	}
+	return nil, nil, nil
+}
+
+// Transform transforms the artifacts
+func (t *PHPDockerfileGenerator) Transform(newArtifacts []transformertypes.Artifact, oldArtifacts []transformertypes.Artifact) ([]transformertypes.PathMapping, []transformertypes.Artifact, error) {
+	pathMappings := []transformertypes.PathMapping{}
+	artifactsCreated := []transformertypes.Artifact{}
+	for _, a := range newArtifacts {
+		if a.Artifact != artifacts.ServiceArtifactType {
+			continue
+		}
+		relSrcPath, err := filepath.Rel(t.Env.GetEnvironmentSource(), a.Paths[artifacts.ProjectPathPathType][0])
+		if err != nil {
+			logrus.Errorf("Unable to convert source path %s to be relative : %s", a.Paths[artifacts.ProjectPathPathType][0], err)
+		}
+		var sConfig artifacts.ServiceConfig
+		err = a.GetConfig(artifacts.ServiceConfigType, &sConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", sConfig, err)
+			continue
+		}
+		sImageName := artifacts.ImageName{}
+		err = a.GetConfig(artifacts.ImageNameConfigType, &sImageName)
+		if err != nil {
+			logrus.Debugf("unable to load config for Transformer into %T : %s", sImageName, err)
+		}
+		var detectedPorts []int32
+		var phpConfig PhpTemplateConfig
+		port, confFileName, err := DetectConf(a.Paths[artifacts.ProjectPathPathType][0])
+		if err != nil {
+			logrus.Debugf("Could not find port in the conf file %s: %s", confFileName, err)
+		} else {
+			if port != 0 {
+				detectedPorts = append(detectedPorts, port)
+			}
+			phpConfig.ConfFile = confFileName
+		}
+		detectedPorts = commonqa.GetPortsForService(detectedPorts, a.Name)
```

Comment:
If the user specifies a port or we choose to use another port, we should consider adding our own conf file to the docker image to configure the apache server to run in that port.

Reply:
okay sure.